lib/slack-ruby-bot/commands/base.rb

```diff
@@ -1,7 +1,69 @@
 module SlackRubyBot
   module Commands
+
+    # class methods to modify the Commands::Base class. All methods are related to
+    # providing before/after hooks to commands.
+    module CommandHooks
```

Comment:
All this code is similar to `ActiveSupport::Callbacks` isn't it? Why not use that.